c7n/query.py

```diff
@@ -540,6 +529,54 @@ def generate_arn(self):
         return self._generate_arn
 
 
+class MaxResourceLimit(object):
+
+    def __init__(self, policy, selection_count, population_count):
+        self.p = policy
+        self.op = None
+        self.selection_count = selection_count
+        self.population_count = population_count
+        self.amount = selection_count
+        self.percentage_amount = population_count
+        self.percent = 100
+        self._parse_policy()
+
+    def _parse_policy(self,):
+        if isinstance(self.p.max_resources, dict):
+            self.op = self.p.max_resources.get("op", "or").lower()
+            self.percent = self.p.max_resources.get("percent", 100)
+            self.amount = self.p.max_resources.get("amount", self.population_count)
+
+        if isinstance(self.p.max_resources, int):
+            self.amount = self.p.max_resources
+
+        if isinstance(self.p.max_resources_percent, (int, float)):
+            self.percent = self.p.max_resources_percent
+
+        self.percentage_amount = self.population_count * (self.percent / 100.0)
+
+    def check_resource_limits(self):
+        if self.selection_count > self.amount and self.selection_count > self.percentage_amount:
+            if self.op == "and":
```

Comment:
let's add the op to initial conditional since there is no other clause on the nested if.